File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.cs
using System;
using System.Collections.Generic;

public class Solution 
{    
    public int[] solution(int l, int r) 
    {
        string str = "";
        
        List<int> numbers = new List<int>();
        
        for (int i=l; i<=r; i++) {
            str = i.ToString();
            bool isValid = true;
            
            foreach (char ch in str) {
                if (ch != '5' && ch != '0') {
                    isValid = false;
                    break;
                }
            }

            if (isValid) {
                numbers.Add(i);
            }
            
        }
        
        if (numbers.Count == 0) {
            numbers.Add(-1);
        }
        
        int[] answer = numbers.ToArray();  
        return answer;
    }
}

File: 프로그래머스/0/181921. 배열 만들기 2/배열 만들기 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int l, int r) 
{
    vector<int> answer;
 
    for(int i=l; i<=r; i++)
    {
        string i_str = to_string(i);
        bool isCheck = true;
        for(char &c : i_str)
        {
            if(c != '5' && c != '0')
            {
                isCheck = false;
            }
        }
        if(isCheck) { answer.push_back(i); }
    }
    if(answer.empty()) { answer.push_back(-1); }
    return answer;
}